### PR TITLE
[core] Minor update on runtime env manager

### PR DESCRIPTION
**Pull request by dentiny**

### CODE REVIEW

File: src/ray/common/runtime_env_manager.cc [hunk 1]
```diff
@@ -29,12 +29,19 @@ void RuntimeEnvManager::AddURIReference(const std::string &hex_id,
                                         const rpc::RuntimeEnvInfo &runtime_env_info) {
   const auto &uris = runtime_env_info.uris();
   if (!uris.working_dir_uri().empty()) {
+    uri_reference_.reserve(uri_reference_.size() + uris.working_dir_uri().size());
```
Comment (rynewang) [inline src/ray/common/runtime_env_manager.cc:32]: If URI already exists, we only want to increment it and this reserve would reserve some unused mem.

File: src/ray/common/runtime_env_manager.cc [hunk 2]
```diff
@@ -29,12 +29,19 @@ void RuntimeEnvManager::AddURIReference(const std::string &hex_id,
...
+    id_to_uris_.reserve(id_to_uris_.size() + uris.working_dir_uri().size());
+
     const auto &uri = uris.working_dir_uri();
     uri_reference_[uri]++;
     id_to_uris_[hex_id].push_back(uri);
     RAY_LOG(DEBUG) << "[working_dir] Added URI Reference " << uri << " for id " << hex_id;
   }
+
   for (const auto &uri : uris.py_modules_uris()) {
+    uri_reference_.reserve(uri_reference_.size() + uris.working_dir_uri().size());
```
Comment (rynewang) [inline src/ray/common/runtime_env_manager.cc:42]: ditto

File: src/ray/common/runtime_env_manager.cc [hunk 3]
```diff
@@ -51,21 +58,23 @@ const std::vector<std::string> &RuntimeEnvManager::GetReferences(
...
-  for (const auto &uri : id_to_uris_[hex_id]) {
-    --uri_reference_[uri];
-    auto ref_count = uri_reference_[uri];
-    RAY_CHECK(ref_count >= 0);
-    if (ref_count == 0) {
-      uri_reference_.erase(uri);
+  for (const auto &uri : iter->second) {
+    auto uri_ref_iter = uri_reference_.find(uri);
+    RAY_CHECK(uri_ref_iter != uri_reference_.end());
+    const auto new_ref_count = --uri_ref_iter->second;
```
Comment (rynewang) [inline src/ray/common/runtime_env_manager.cc:69]: I don't like -- and ++ with assignments. Let's put it 2 lines

Comment (dentiny) [reply src/ray/common/runtime_env_manager.cc:69]: > I don't like -- and ++ with assignments

Any reasoning? Coding practice / suggestion?

Comment (rynewang) [reply src/ray/common/runtime_env_manager.cc:69]: code readability. when I read 
```
auto y = ++x;
```

I need to reason: okay so y is assigned to the value before increment, not after. 

but when I read

```
auto y = x;
++x;
```

there's no such cognitive load.

Comment (rynewang) [reply src/ray/common/runtime_env_manager.cc:69]: wait I even made a mistake, it's after the increment not before lol. this really makes the point.

File: src/ray/common/runtime_env_manager.cc (hunk 1 above)
Comment (dentiny) [reply src/ray/common/runtime_env_manager.cc]: You're right, removed.

File: src/ray/common/runtime_env_manager.cc (hunk 2 above)
Comment (dentiny) [reply src/ray/common/runtime_env_manager.cc]: You're right, removed.

File: src/ray/common/runtime_env_manager.cc (hunk 3 above)
Comment (dentiny) [reply src/ray/common/runtime_env_manager.cc]: I think this is fundamental C/C++ grammar, not something fancy and hard-to-understand.
But updated to make you happy.

**Review by jjyao — APPROVED** [review]

**Pull request merged**